docs/source/gsoc.rst

```diff
@@ -0,0 +1,57 @@
+
+Working on LiberTEM will give you experience in developing distributed systems for high-performance data processing with Python. You can learn how to do profiling and targeted performance optimization: Our current records are an aggregate of 49 GB/s on eight low-end microblade nodes reading from mass storage, and 21 GB/s on a single high-end workstation reading from the file system cache. 
+
+If you work on our GUI, you'll learn how a responsive web application for big data analytics can be built with a front-end based on TypeScript, React and Redux, and an asynchronous Python back-end based on on Tornado and dask.distributed. 
+
+Our team is experienced in software development methodology and tools, product management, product development, open source business models and strategy, and start-ups. We'd be happy to share our experience with you!
+
+How to reach us
+---------------
+
+The easiest path is our Gitter channel: https://gitter.im/LiberTEM/Lobby
+
+E-Mail: `Dieter Weber d.weber@fz-juelich.de <mailto:d.weber@fz-juelich.de>`_
+
+Just drop a message! We are based in Germany and are generally active during the day. :-)
+
+Getting Started
+---------------
+
+If you have questions, please ask freely: Supporting users and constributors has a high priority for us. Our development is currently moving very quickly. We are planning to complete our documentation when a few major features have left the prototype stage. Until then it is always a good idea to just ask if you are missing information.
+
+Installation
+~~~~~~~~~~~~
+
+Please see `our documentation <https://libertem.github.io/LiberTEM/install.html>`_ for details! Forking our repository, cloning the fork and `installing the clone <https://libertem.github.io/LiberTEM/install.html#installing-from-a-git-clone>`_ are the recommended setup if you will be contributing significant amounts of code.
+
+Currently, we are still working on getting suitable sample files online. Please contact us to get sample data!
+
+
+
+What's the process for submitting your first bug fix?
+Where should students look to find easy bugs to try out?
+Writing your GSoC application
+Links to advice about applications and the application template goes here.
+
+Remind your students that your sub-org name must be in the title of their applications!
+Here's a link to the student application information for Python
+Project Ideas
+You should usually have a couple of project ideas, ranging in difficulty from beginner to expert. Please do try to have at least one, preferably several beginner tasks: GSoC gets a lot of students with minimal open source experience who feel very discouraged (and sometimes even complain to Google) if orgs don't any have projects at their level.
```

Comment:
Hmm, this is crucial: we need to find something that that is easy enough, yet leaves the student with a) enough to do for three months and b) some sense of accomplishment. Maybe we can think of some UDFs that are useful yet easy to implement? 

On the more complicated side, some ideas/areas:

- Proper schemas, validation and automatic form generation for analysis parameters. This is a cross-section through Python and TypeScript, though we could also split off the more react-y part. Does not require numpy knowledge, or domain knowledge, Python/TypeScript required. General WebDev experience could help.
- Explore automated benchmarks in detail. Run Benchmarks in CI and observe variance. Create synthetic test data. Record and present benchmark results over time
- Compression survey. Analyse high-throughput compression techniques, dive into lz4/zstd, blosc etc, compare against existing file formats, ... - some low-level programming experience would be of advantage. can be done basically independent of the LiberTEM codebase.
- Maybe something Operations/Cloud/Kubernetes/... related? Our caching layer?
- A completely new analysis. Maybe this could be another avenue for implementing something like RDF mapping, a challenging project for an advanced student, who likes to do a deep dive into a topic. Python/numpy knowledge should be required, to have more time to actually work on the problem at hand. We can have several of these. User interface can be included, at least a concept for the workflow can be developed, depending on ambitions of the student.
- Live visualization of large binary data. Basically an efficient/zoomable/user-friendly/fully-featured replacement for our visualization. Requires a cross-section of different technologies from Python/numpy/threading over HTTP/websockets to Canvas/WebGL. Could be spun off into its own project if it is successful!

How should we present our ideas? I guess there are some requirements from GSOC for this? I guess this: https://google.github.io/gsocguides/mentor/defining-a-project-ideas-list

Reply:
Thx @sk1p , I've built the ideas list based on your items, see a344440! :-)